### PR TITLE
Add initial documentation for ClickPipes Avro beta

### PR DESCRIPTION
**Pull request by genzgd**
Small docs update to describe Avro for ClickPipes.  We should expand this but preferably in connection with some UI updates for the schema registry/discovery steps.

### CODE REVIEW

**Comment by pjhampton** [comment]
Are you merging to main, or just sharing the preview link for the beta?

**Comment by genzgd** [comment]
How do I share the preview link?  I assume we shouldn't merge to main until we remove the feature flag gate.

**Comment by pjhampton** [comment]
Here is the preview link: https://clickhouse-docs-jbejl7xos-clickhouse.vercel.app/docs/en/integrations/clickpipes
If you scroll up there is a button that say `View Deployment`

> I assume we shouldn't merge to main until we remove the feature flag gate.

We aren't merging to `main` until S3 GA. I'm not sure if you want to follow a similar practice?

**Comment by genzgd** [comment]
Ah, thank you, I never noticed that magic button before (but it's been a while since I've done any docs).  I'll share that as appropriate, and I also have some thoughts about adding some additional detail for Avro.

File: docs/en/integrations/data-ingestion/clickpipes/index.md [hunk 1]
```diff
@@ -144,6 +143,19 @@ Nullable versions of the above are also supported with these exceptions:
...
+## Avro Schema Management
+
+ClickPipes dynamically retrieves and applies the Avro schema from the configured Schema Registry using the schema ID embedded in each message/event.  Schema updates are detected and processed automatically.
+
+The following rules are applied to the mapping between the retrieved Avro schema and the ClickHouse destination table:
+- If the Avro schema contains a field that is not included in the ClickHouse destination mapping, that field is ignored.
+- If the Avro schema is missing a field defined in the ClickHouse destination mapping, the ClickHouse column will be populated with a "zero" value, such as 0 or an empty string.  Note that `DEFAULT` expressions are not currently evaluated for ClickPipes inserts (this is temporary limitation pending updates to the ClickHouse server default processing).
+- If the Avro schema field and the ClickHouse column are incompatible, inserts of that row/message will fail and the failure will be recorded in the ClickPipes errors table.  Note that several implicit conversions are supported (like between numeric types), but not all (for example, an Avro `record`field can not be inserted into an `Int32` ClickHouse column).
```
Comment (mshustov) [inline docs/en/integrations/data-ingestion/clickpipes/index.md:157]:
```suggestion
- If the Avro schema field and the ClickHouse column are incompatible, inserts of that row/message will fail, and the failure will be recorded in the ClickPipes errors table.  Note that several implicit conversions are supported (like between numeric types), but not all (for example, an Avro `record` field can not be inserted into an `Int32` ClickHouse column).
```

File: docs/en/integrations/data-ingestion/clickpipes/index.md [hunk 2]
```diff
@@ -40,8 +40,7 @@ Currently ClickPipes does not support loading custom CA certificates.
...
 
-4a. __Optional for Schema Registry__: Specify the complete URL of your Schema Registry server along with the precise RESTful path to the ID representing your preferred schema document. This schema will serve as the validation benchmark for all messages from your topic, consequently blocking any messages that fail the validation against the designated schema. Additionally, the Schema Registry allows new schema retreival via JSON_SR messages. In scenarios where your JSON_SR message carries a Schema Registry ID differing from the current one, Clickpipes will fetch the corresponding schema and use it for future validation.
-
+4a. __Schema Registry__: A valid schema is required for Avro streams and optional for JSON. Specify the complete URL of your Schema Registry server along with the full path to the ID of the relevant schema document.  This schema will be used to parse (Avro) or validate (JSON) messages on the selected topic.  Avro messages that can not be parsed or JSON messages that fail validation will generate an error.  Note that ClickPipes will automatically retrieve an updated or different schema from the registry if indicated by the schema ID embedded in the message.
```
Comment (mshustov) [inline docs/en/integrations/data-ingestion/clickpipes/index.md:43]:
```suggestion
4a. __Schema Registry__: A valid schema is required for Avro streams and optional for JSON. Specify the complete URL of your Schema Registry server along with the full path to the ID of the relevant schema document.  This schema will be used to parse ([AvroConfluent](../../../interfaces/formats.md/#data-format-avro-confluent)) or validate (JSON) messages on the selected topic.  Avro messages that can not be parsed or JSON messages that fail validation will generate an error.  Note that ClickPipes will automatically retrieve an updated or different schema from the registry if indicated by the schema ID embedded in the message. 
```

File: docs/en/integrations/data-ingestion/clickpipes/index.md [hunk 3]
```diff
@@ -106,13 +105,13 @@ The supported formats are:
 | Format                                                                                    | Support     |
 |-------------------------------------------------------------------------------------------|-------------|
 | [JSON](../../../interfaces/formats.md/#json)                                               | ✔           |
-| [AvroConfluent](../../../interfaces/formats.md/#data-format-avro-confluent)                |*Coming Soon*|
+| [AvroConfluent](../../../interfaces/formats.md/#data-format-avro-confluent)                |✔|
```
Comment (mshustov) [inline docs/en/integrations/data-ingestion/clickpipes/index.md:108]:
```suggestion
| [AvroConfluent](../../../interfaces/formats.md/#data-format-avro-confluent)                |✔ in `beta`, contact support to enable|
```

**Review by mshustov — APPROVED** [review]
I'd suggest marking `Avro` as available in `beta` and merging the docs. The more customers see what's supported, the more testers we might get.

**Pull request merged**